tests/unit/model-data-test.js

```diff
@@ -184,14 +185,26 @@ module('unit/model-data', function(hooks) {
       };
       this.grandchild1ModelData = new M3ModelData(
         'com.exmaple.bookstore.book',
-        'grandchild1_1',
+        'child1_1',
         null,
         this.storeWrapper,
         this.child1ModelData,
         'child1_1',
         false,
         { record: this.grandchild1Model }
       );
+
+      SchemaManager.registerSchema({
+        computeNestedModel(key, value) {
+          if (value !== null && typeof value === 'object') {
+            return { id: key, type: 'com.exmaple.bookstore.book', attributes: value };
+          }
+        },
+      });
+    });
+
+    hooks.afterEach(function() {
+      SchemaManager.registerSchema(null);
     });
 
     test('.pushData calls reified child model datas recursively', function(assert) {
```

Comment:
update test name; it's no longer true.

We only call things recursively for child model datas whose id & type remain unchanged right?

Reply:
Yes, this is correct, but the test does not really tests this - for all child model datas, the type and id remains unchanged. Therefore, I'm not sure whether there is point to clarify this in this test.

There are separate tests in `model-test` to test this behavior.